index.bs

```diff
@@ -998,167 +1039,123 @@ the server, and sending and receiving datagrams.
   } catch (e) {
-    addToEventLog('Error while reading datagrams: ' + e, 'error');
+    addToEventLog(&#96;Error while reading datagrams: ${e}&#96;, 'error');
   }
 }
 
-async function acceptUnidirectionalStreams(transport) {
-  let reader = transport.incomingUnidirectionalStreams.getReader();
+async function acceptUnidirectionalStreams() {
   try {
-    while (true) {
-      const { value, done } = await reader.read();
-      if (done) {
-        addToEventLog('Done accepting unidirectional streams!');
-        return;
-      }
-      let stream = value;
-      let number = streamNumber++;
-      addToEventLog('New incoming unidirectional stream #' + number);
+    for await (stream of transport.incomingUnidirectionalStreams) {
+      const number = streamNumber++;
+      addToEventLog(&#96;New incoming unidirectional stream #${number}&#96;);
       readFromIncomingStream(stream, number);
     }
+    addToEventLog('Done accepting unidirectional streams!');
   } catch (e) {
-    addToEventLog('Error while accepting streams: ' + e, 'error');
+    addToEventLog(&#96;Error while accepting streams ${e}&#96;, 'error');
   }
 }
 
 async function readFromIncomingStream(stream, number) {
-  let decoder = new TextDecoderStream('utf-8');
-  let reader = stream.readable.pipeThrough(decoder).getReader();
   try {
-    while (true) {
-      const { value, done } = await reader.read();
-      if (done) {
-        addToEventLog('Stream #' + number + ' closed');
-        return;
-      }
-      let data = value;
-      addToEventLog('Received data on stream #' + number + ': ' + data);
+    const decoder = new TextDecoderStream('utf-8');
+    for await (chunk of stream.readable.pipeThrough(decoder).getReader()) {
+      addToEventLog(&#96;Received data on stream #${number}: ${chunk}&#96;);
     }
+    addToEventLog(&#96;Stream #${number} closed&#96;);
   } catch (e) {
-    addToEventLog(
-        'Error while reading from stream #' + number + ': ' + e, 'error');
-    addToEventLog('    ' + e.message);
+    addToEventLog(&#96;Error while reading from stream #${number": ${e}&#96;, 'error');
+    addToEventLog(&#96;    ${e.message}&#96;);
   }
 }
 
 function addToEventLog(text, severity = 'info') {
-  let log = document.getElementById('event-log');
-  let mostRecentEntry = log.lastElementChild;
-  let entry = document.createElement('li');
+  const log = document.getElementById('event-log');
+  const last = log.lastElementChild;
+  const entry = document.createElement('li');
   entry.innerText = text;
-  entry.className = 'log-' + severity;
+  entry.className = &#96;log-${severity}&#96;;
   log.appendChild(entry);
 
-  // If the most recent entry in the log was visible, scroll the log to the
-  // newly added element.
-  if (mostRecentEntry != null &&
-      mostRecentEntry.getBoundingClientRect().top <
-          log.getBoundingClientRect().bottom) {
+  // If the last entry in the log is visible, scroll the log to the new element.
```

Comment:
If the most recent entry in the log is not visible, ...

Reply:
I renamed it to `last` for brevity in line 1150 above, otherwise the logic is unchanged, except I see present tense snuck in there, which doesn't help. Let me find a better name...